libblkid/src/superblocks/hfs.c

```diff
@@ -238,17 +242,23 @@ static int probe_hfsplus(blkid_probe pr, const struct blkid_idmag *mag)
 	    (memcmp(hfsplus->signature, "HX", 2) != 0))
 		return 1;
 
-	hfs_set_uuid(pr, hfsplus->finder_info.id, sizeof(hfsplus->finder_info.id));
-
+	/* Block size must be at least as large as a sector and a multiple of 2. */
 	blocksize = be32_to_cpu(hfsplus->blocksize);
-	if (blocksize < HFSPLUS_SECTOR_SIZE)
+	if (blocksize < HFSPLUS_SECTOR_SIZE || ((blocksize - 1) & blocksize))
```

Comment:
`blocksize % 2` would be easier to understand.

Reply:
Copy & past from kernel: https://github.com/torvalds/linux/blob/master/fs/hfsplus/wrapper.c#L234, the comment needs fix.

Reply:
`is_power_of_2()` from `c.h` would be clearer then.

I can send a kernel patch if you prefer.